### PR TITLE
Added isOwner policy

### PR DESCRIPTION
**Pull request by Tenkir**
Checks if the user is the creator of the target object, or is an admin.Returns a 401 or 403 error if not.

### CODE REVIEW

File: backend/api/policies/isOwnerorAdmin.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  var modelType = req.options.model;
+  sails.models[modelType]
+    .findOne(req.query.id)
+    .exec(function(error, model) {
+      if(error) {
+        next(error);
+      }
+
+      if(req.token != model.user) {
```
Comment (tarlepp) [inline backend/api/policies/isOwnerorAdmin.js]: Small fix + style change.

```if (req.token !== model.user) {```

File: backend/api/policies/isOwnerorAdmin.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+              next();
+            } else {
+              error = new Error();
+
+              error.status = 403;
+              error.message = 'Forbidden - You are not an authorized user.';
+              return res.json(error);
+            }
+          });
+      }
```
Comment (tarlepp) [inline backend/api/policies/isOwnerorAdmin.js]: Style change here too.

```
if (...) {
  ...
} else {
```

File: backend/api/policies/isOwnerorAdmin.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          .exec(function exec(error, user) {
+            if (error) {
+              next(error);
+            } else if (!user) {
+              error = new Error();
+
+              error.status = 401;
+              error.message = 'User not found - Please login.';
+
+              return res.json(error);
```
Comment (tarlepp) [inline backend/api/policies/isOwnerorAdmin.js]: ```res.json``` => ```res.negoatiate```

File: backend/api/policies/isOwnerorAdmin.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            } else if (user.admin) {
+              next();
+            } else {
+              error = new Error();
+
+              error.status = 403;
+              error.message = 'Forbidden - You are not an authorized user.';
+              return res.json(error);
```
Comment (tarlepp) [inline backend/api/policies/isOwnerorAdmin.js]: Same as above

**Comment by tarlepp** [comment]
This one I really like, thanks! But there is some minor changes that I would like to see before I accept this one.

* Change ```.exec()``` to use promises
* Handle errors within promise ```.catch()```
* Add policy to all ```update``` routes
* Use sails ```ActionUtil``` to parse ```model``` and ```id``` [example](https://github.com/balderdashy/sails/blob/master/lib/hooks/blueprints/actions/update.js#L22-L26)
* Check this [style guide](https://github.com/airbnb/javascript/tree/master/es5) for code format

Also I'm thinking of ```isAdmin``` policy that could be attached to ```create``` routes. And another thing is assoc handling ([add to](http://sailsjs.org/documentation/reference/blueprint-api/add-to) and [remove from](http://sailsjs.org/documentation/reference/blueprint-api/remove-from)) with this policy or a new one.

What are thoughs about my comments?

**Comment by tarlepp** [comment]
Any progress with this?

**Comment by tarlepp** [comment]
Any progress with this?

**Comment by tarlepp** [comment]
Any progress with this?

**Comment by tarlepp** [comment]
Any progress with this? Going to close this one if replies won't come.

**Comment by Tenkir** [comment]
Well this PR needs to be closed anyways since the front and backend have been moved into submodules.

**Pull request closed without merging**